sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -284,6 +284,7 @@ export class EventHubClient {
    * @returns Promise<void>
    */
   createProducer(options?: EventHubProducerOptions): EventHubProducer {
+    throwErrorIfConnectionClosed(this._context);
```

Comment:
Should we add a `@throws` section to the JSDOC for this error? Maybe add something like "Create a new EventHubsClient and call createProducer when encountering this error"